### PR TITLE
slop: add missing libSM dependency

### LINKED ISSUE #182568: `maim` broken since f7414e2b3ac69f220e19b1ef9ddfe8d64650c5ca (potentially more)
### Describe the bug

`maim` errors with "/nix/store/y6w2r99psf1cznq179kyq2mzk1zyh6xn-maim-5.7.4/bin/maim: error while loading shared libraries: libSM.so.6: cannot open shared object file: No such file or directory" (or equivalent) since f7414e2b3ac69f220e19b1ef9ddfe8d64650c5ca which was a cmake/rpath related change.

As the commit was a treewide operation, there might actually be much more packages affected, though `maim` happened to be the first one reported in the inofficial discord.

### Steps To Reproduce
1. `nix run github:nixos/nixpkgs/f7414e2b3ac69f220e19b1ef9ddfe8d64650c5ca#maim -- --version`

### Expected behavior

Maim reports its version, and is generally usable

### Notify maintainers

@knedlsepp for being the author of the offending commit, sadly maim has no maintainers listed on its own.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix run nixpkgs#nix-info -- -m
 - system: `"x86_64-linux"`
 - host os: `Linux 5.18.9, NixOS, 22.11 (Raccoon), 22.11.20220714.4a01ca3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.10.0pre20220715_59764eb`
 - channels(nmelzer): `""`
 - channels(root): `"nixos-21.05.4044.e74894146a4"`
 - nixpkgs: `/etc/nixpkgs/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by Kranzes**
###### Description of changes
- Added a missing dependency called libSM
- Formatted the file modified files `nixpkgs-fmt`

Closes #182568

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Kranzes** [comment]
Result of `nixpkgs-review pr 182506` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>maim</li>
    <li>pb_cli</li>
    <li>yaxg</li>
  </ul>
</details>

**Comment by reedrw** [comment]
slop was also recently broken by libSM dependency. Since maim depends on slop, I think it would make sense to also fix it for slop in this PR. In fact, I suspect fixing the dependency for slop will also fix it for maim.

**Comment by Kranzes** [comment]
Result of `nixpkgs-review pr 182506` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>capture</li>
    <li>giph</li>
    <li>maim</li>
    <li>pb_cli</li>
    <li>recapp</li>
    <li>slop</li>
    <li>yaxg</li>
  </ul>
</details>

File: pkgs/tools/misc/slop/default.nix [hunk 1]
```diff
@@ -14,17 +25,28 @@ stdenv.mkDerivation rec {
...
+    glew
+    glm
+    libGLU
+    libGL
+    libX11
+    libXext
+    libXrender
+    icu
+    libSM
+  ];
```
Comment (smancill) [inline pkgs/tools/misc/slop/default.nix:43]: `cppcheck` was removed from the inputs, it should be explained in the commit message why.

Comment (Kranzes) [reply pkgs/tools/misc/slop/default.nix:43]: I removed it because the build went totally fine without it, does that seem right?

**Pull request merged**